ignite/engine/engine.py

```diff
@@ -252,8 +247,9 @@ def print_epoch(engine):
             See :class:`~ignite.engine.Events` for more details.
 
         """
-        if isinstance(event_name, EventWithFilter):
-            event_name, event_filter = event_name.event, event_name.filter
+        if (isinstance(event_name,
+                       CallableEventWithFilter) and event_name.filter != CallableEventWithFilter.default_event_filter):
```

Comment:
@justusschock I can be wrong, but can't we just set `event_name.filter = None` for default non filtered behaviour ?
I do not see where `CallableEventWithFilter.default_event_filter` is actually used except comparision...

Reply:
We can't, since we explicitly call the filter without additional typechecks.